Code/Framework/AzFramework/CMakeLists.txt

```diff
@@ -10,6 +10,18 @@
 ly_get_list_relative_pal_filename(pal_dir ${CMAKE_CURRENT_LIST_DIR}/Platform/${PAL_PLATFORM_NAME})
 ly_get_list_relative_pal_filename(common_dir ${CMAKE_CURRENT_LIST_DIR}/Platform/Common)
 
+ly_add_target(
+    NAME AzFramework.Headers
+    NAMESPACE AZ
+    HEADERONLY
+    INCLUDE_DIRECTORIES
+        INTERFACE
+            .
+            ${pal_dir}
+)
```

Comment:
Can this be moved into Code/Framework/AzFramework/Platform/Common/Xcb/CMakeLists.txt (or Code/Framework/AzFramework/Platform/Linux/platform_linux.cmake)? Doesn't seem like it's needed for any other platform.

Reply:
One objective here is to be forward-thinking to the Wayland support. Wayland will need this as well. But that's not the only issue.

`Code/Framework/AzFramework/Platform/Linux/platform_linux.cmake` is not just any old cmake file. It is used as a `PLATFORM_INCLUDE_FILES` argument in a call to `ly_add_target`, and is expected to set (or not set) specific variables (like `LY_FILES`, `LY_INCLUDE_DIRECTORIES`, etc). It is meant to configure the current target being created by `ly_add_target`. It is included by the CMake `include()` command, which:
* does not open a new scope
* does not affect `${CMAKE_CURRENT_LIST_DIR}`
Because it does not open a new scope, calling `ly_add_target` from within a platform include file amounts to calling `ly_add_target` from within `ly_add_target`, and the scopes of the variables gets really mixed up. Variables from the second call end up polluting the state of the first call.

Because of this scoping issue, you'll notice that `Code/Framework/AzFramework/Platform/Linux/platform_linux_window_system.cmake` calls `add_subdirectory()` as opposed to `include()`. `add_subdirectory()`:
* opens a new scope
* changes `${CMAKE_CURRENT_LIST_DIR}`.
Because `add_subdirectory()` changes `${CMAKE_CURRENT_LIST_DIR}`, now the issue is that the Xcb code needs to know the include path to the main AzFramework code. We can either:
* Have `AzFramework/CMakeLists.txt` set some variables to say where the AzFramework root directory and pal directories are, and use those in `Xcb/CMakeLists.txt`
* Have `AzFramework/CMakeLists.txt` create a target with the right properties, and use that target in `Xcb/CMakeLists.txt`
Using targets is the more modern CMake approach, so that's what I went with.

Reply:
Wayland is also Linux specific though, so would be good if all of this could be confined to the linux specific cmake files in order to avoid having to add a bunch of empty boilerplate stuff for every other platform that doesn't need it.

Not sure if it's the 'right' way to do it, but I've added new platform specific cmake targets inside of 'Platform/{Platform}/platform_{platform}.cmake files before and it works, although maybe I just got lucky:)

In either case not a big deal.

Reply:
This all happens in a platform include file now.